File: HBV_genotype_coverage.R
##Author: ZHU O. Yuan 2017
##Script name: HBV_genotype_coverage.R
##Rscript to generate coverage plot and SNP locations
##Usage: Rscript HBV_genotype_coverage.R $ID $MEAN $COV $VCF $Rout
##Example output pdf see HBVmap-example-pdf_AHB772_B...pdf at https://github.com/YuanOZhu/HBVGenotypeAwareMapper

args<-commandArgs(TRUE)
id<-args[1] #sample name
aveCov<-args[2] #sample mean coverage

pdf(args[5]) #pdf file name

data<-read.table(file=args[3],sep="\t",header=TRUE) #coverage file
data_pos<-do.call(rbind,strsplit(as.character(data[,1]),":"))
header<-paste(id," Mean Coverage: ", aveCov, sep="")
#plot log coverage against base position
plot(as.numeric(data_pos[,2]),log10(data[,2]),main=header,ylab="Log10(cov)",xlab="Pos",ylim=c(0,7),pch=20) 

vcf<-read.table(file=args[4],sep="\t",header=FALSE) #vcf file
vcf_filter<-vcf[which(vcf[,6]>1000),]
vcf_data<-do.call(rbind,strsplit(as.character(vcf_filter[,8]),";"))
vcf_freq<-do.call(rbind,strsplit(as.character(vcf_data[,2]),"="))
vcf_dp<-do.call(rbind,strsplit(as.character(vcf_data[,1]),"="))
#plot SNP frequency against base position
points(vcf_filter[,2],as.numeric(vcf_freq[,2])*max(log10(data[,2])),col="red",pch=20)
dev.off()

File: HBV_genotype_coverage.pl
use warnings;
use strict;

##Author: ZHU O. Yuan 2017
##Script name: HBV_genotype_coverage.pl
##Coverage plot generating script called upon by HBV_genotype_mapper.sh
##Usage: perl Plot_coverage.pl AHB001_B path_to_vcf
##Calls: Rscript HBV_genotype_coverage.R

my$ID=$ARGV[0];#sample ID
my$MEAN=0; #average coverage
my$COV=$ARGV[0].".sorted.grouped.realign.coverage"; #coverage file
my$AVECOV=$ARGV[0].".sorted.grouped.realign.coverage.sample_summary"; #coverage summary file
my$VCF=$ARGV[1].$ARGV[0].".sorted.grouped.realign.recal.lofreq.vcf"; #vcf file
my$Rout=$ARGV[0].".sorted.grouped.realign.recal.lofreq.coverage.pdf"; #out pdf

#calculates average coverage across genome for input into R script
open (SAM, "$AVECOV") or die; 
while (my $line = <SAM>) {
    chomp $line;
    my@data=split('\t',$line);
    if($data[0] eq $ID){$MEAN=$data[2];}
}close (SAM);

#calls R script to generate coverage plot
system("Rscript HBV_genotype_coverage.R $ID $MEAN $COV $VCF $Rout");
